src/main/app/controller/courts/PostcodesController.ts

```diff
@@ -0,0 +1,220 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {PostcodeData} from '../../../types/Postcode';
+import {Error} from '../../../types/Error';
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+import {AreaOfLaw} from '../../../types/AreaOfLaw';
+import {CourtType} from '../../../types/CourtType';
+import {familyAreaOfLaw} from '../../../enums/familyAreaOfLaw';
+
+@autobind
+export class PostcodesController {
+
+  getCourtTypesErrorMsg = 'A problem occurred when retrieving the court types.';
+  getCourtAreasErrorMsg = 'A problem occurred when retrieving the court areas of law.';
+  getCourtAreasOfLawErrorMsg = 'A problem occurred when retrieving the court areas of law. ';
+  familyAreaOfLawErrorMsg = 'You need to enable relevant county court areas of law: Housing, Money Claims, or Bankruptcy';
+  getPostcodesErrorMsg = 'A problem occurred when retrieving the postcodes.';
+  addErrorMsg = 'A problem has occurred (your changes have not been saved). The following postcodes are invalid: ';
+  deleteErrorMsg = 'A problem has occurred when attempting to delete the following postcodes: ';
+  moveErrorMsg = 'A problem has occurred when attempting to move the following postcodes: ';
+  moveErrorDuplicatedMsg = 'The postcode is already present on the destination court: ';
+  postcodesNotValidMsg = 'The postcode provided needs to be more than one character up to the full length of a postcode: '
+  noPostcodeErrorMsg = 'Please update the required form below and try again.'
+  duplicatePostcodeMsg = 'One or more postcodes provided already exist: ';
+  noSelectedPostcodeMsg = 'Please select one or more postcodes to delete.'
+  noSelectedPostcodeOrCourtMsg = 'Please select one or more postcodes and a court before selecting the move option.'
+
+  public async get(
+    req: AuthedRequest,
+    res: Response,
+    searchValue = '',
+    postcodes: string[] = null,
+    error = '',
+    areasOfLaw: string[] = null,
+    courtTypes: string[] = null,
+    updated = false): Promise<void> {
+    const slug: string = req.params.slug as string;
+
+    const errors: Error[] = [];
+    // If we have an error from validation when adding/removing or moving postcodes,
+    // append it
+    if (error) {
+      errors.push({text: error});
+    }
+
+    if (!areasOfLaw) {
+      await req.scope.cradle.api.getCourtAreasOfLaw(slug)
+        .then((value: AreaOfLaw[]) => areasOfLaw = value.map(ct => ct.name))
+        .catch(() => errors.push({text: this.getCourtAreasOfLawErrorMsg}));
+    }
+
+    if (areasOfLaw) {
+      areasOfLaw = this.checkCountyAreasOfLaw(areasOfLaw).map(ct => ct.replace(/\s/g, '_'));
```

Comment:
The checkCountyAreasOfLaw replaces underscores with spaces, then this replaces spaces with underscores? Is that needed?

Reply:
I think I put a comment further down, but the input fields cut out half way through the string array if spaces are included! A bit of a pain, so it is needed unfortunately